UNITTESTS/stubs/platform/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-stubs-platform)
+
+target_sources(mbed-stubs-platform
+    PRIVATE
+        mbed_critical_stub.c
+        mbed_atomic_stub.c
+        mbed_error.c
+        mbed_poll_stub.cpp
+        mbed_assert_stub.cpp
+        mbed_wait_api_stub.cpp
+        mbed_retarget_stub.cpp
+        FileHandle_stub.cpp
+        nvic_wrapper_stub.c
+        randLIB_stub.c
+        randLIB_stub.cpp
+)
+
+target_link_libraries(mbed-stubs-platform
+    PRIVATE
+        mbed-headers
+        mbed-stubs-headers
+        gcov
```

Comment:
As above, this won't work for AppleClang. We need to pass target_link_options(mbed-stubs-platform --coverage) instead. That will alias to the correct linker argument on gcc and AppleClang. The current setup causes a build failure with the error `library not found for -lgcov` when building with AppleClang.